prow/cmd/deck/main.go

```diff
@@ -1488,7 +1696,12 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool, cfg
 			}
 			return
 		}
-		newPJ := pjutil.NewProwJob(pj.Spec, pj.ObjectMeta.Labels, pj.ObjectMeta.Annotations)
+		newPJ, err := newRerunProwJob(mode, cfg, pj)
```

Comment:
the mode switch is a major business logic, imo it would be better handled here instead of by the `newRerunProwJob` function

Reply:
Fixed!